### PR TITLE
Move database cleanup routine into DatabaseCleaner type

### PR DESCRIPTION
**Pull request by mplorentz**
## Issues covered
This is a quick refactoring PR so I can work on the solution to problem 1 in https://github.com/planetary-social/nos/issues/703#issuecomment-2147829931
> The database cleanup script is deleting EventReferences and replacing them with new ones, breaking SwiftUI observation. Solution: refactor EventReferences out of model layer.

## Description
This refactors the guts of the database cleanup routine out of `PersistenceController` and into a new `DatabaseCleaner` type. I also removed some protections we had around calling the routine simultaneously from different threads, because even with the protections it is isn't safe to call this more than once, or at any time other than right at app launch because notes you are looking at could disappear in front of your eyes. I also stubbed out a test file that I will use to TDD my next PR. The app should function exactly as it did before after this PR is merged.

## How to test
The best way to observe the behavior of the cleanup routine is to filter the logs for "Database statistics" and see how the numbers change.

### CODE REVIEW

File: NosTests/Service/DatabaseCleanerTests.swift [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import XCTest
+
+final class DatabaseCleanerTests: CoreDataTestCase {
+    
+    func test_emptyDatabase() async throws {
+        try await DatabaseCleaner.cleanupEntities(before: Date.now, for: KeyFixture.alice.publicKeyHex, in: testContext)
```
Comment (joshuatbrown) [inline NosTests/Service/DatabaseCleanerTests.swift:6]: Do we need an assert in here? Or is this just to verify that cleanupEntities doesn't crash?

**Review by joshuatbrown — COMMENTED** [review]
Looks great and works well. One question.

File: NosTests/Service/DatabaseCleanerTests.swift (hunk 1 above)
Comment (mplorentz) [reply NosTests/Service/DatabaseCleanerTests.swift:6]: Yeah this is just to make sure it doesn't crash. I suppose we could assert that cleanupEntities didn't _create_ any entities but that seems a little crazy. But why not, I can do that quickly.

**Review by martindsq — APPROVED** [review]
Looks good to me. Its seems fine to be merged as Josh already expressed his 👌

**Pull request merged**